### PR TITLE
docs(FilterPanelCheckboxWithOverflowMenu): implement selectorPrimaryFocus instead of primaryFocus

### PR DESCRIPTION
**Pull request by jendowns**
## Affected issues

partially addresses #683 

## Proposed changes

- update `OverflowMenuItem` in `FilterPanelCheckboxWithOverflowMenu` story to implement `selectorPrimaryFocus` prop

### CODE REVIEW

**Comment by diego-codes** [comment]
Does the UI work as expected without the prop?

**Comment by jendowns** [comment]
@diego-codes This PR is for a storybook demo.

**Comment by diego-codes** [comment]
I know, I was just curious because it's unclear to me what the purpose of the prop is. So if the prop is totally optional, then would there be a point in keeping it for the previews which are what we rely on for documentation.

File: src/components/FilterPanel/FilterPanelCheckboxWithOverflowMenu/FilterPanelCheckboxWithOverflowMenu.stories.js [hunk 1]
```diff
@@ -28,13 +28,13 @@ storiesOf(
...
       <OverflowMenuItem
-        primaryFocus
         itemText="Option 1"
         onClick={action('OverflowMenuItem onClick')}
       />
       <OverflowMenuItem
-        itemText="Option 2"
+        itemText="Option 2 -- selected when menu opens"
         onClick={action('OverflowMenuItem onClick')}
+        data-floating-menu-primary-focus
```
Comment (diego-codes) [inline src/components/FilterPanel/FilterPanelCheckboxWithOverflowMenu/FilterPanelCheckboxWithOverflowMenu.stories.js:37]: This isn't a prop that is documented, does it make sense to have it?

Comment (jendowns) [reply src/components/FilterPanel/FilterPanelCheckboxWithOverflowMenu/FilterPanelCheckboxWithOverflowMenu.stories.js:37]: It's the default value for `selectorPrimaryFocus` in `OverflowMenu`

**Pull request merged**